test/topology_custom/test_tablets_cql.py

```diff
@@ -66,3 +67,65 @@ async def alter_tablets_ks_without_waiting_to_complete():
 
     with pytest.raises(InvalidRequest, match="Can't ALTER keyspace ks, keyspace doesn't exist") as e:
         await task
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_alter_tablets_keyspace_concurrent_modification(manager: ManagerClient) -> None:
```

Comment:
I think both with this and the test above you are not using the right framework for the job, those topology tests are not unittests for this particular piece of the code, you're also adding pretty specific injection points which also suggests that there is something odd about this way of testing.

Why this wasn't implemented in test/boost/tablets_test.cc instead?

Reply:
> you're also adding pretty specific injection points

That's because to reproduce the problem a specific interleaving of events is required across multiple servers. The injections are used to orchestrate this order of events.

Reply:
If we do this for every bug soon injection code will take over the source and you will have more injections than actual code.

Reply:
I wish we tested our code that well. We are nowhere near such situation and I doubt we'll ever be.

Reply:
> If we do this for every bug soon injection code will take over the source and you will have more injections than actual code.

We may as well remove the testcase, I don't see better solutions. I assume we want to leave the testcase.